src/main/java/marquez/resources/JobResource.java

```diff
@@ -1,26 +1,55 @@
 package marquez.resources;
 
-import java.util.List;
+import static marquez.resources.ResourceUtil.buildLocation;
+
+import com.codahale.metrics.annotation.Timed;
 import javax.ws.rs.Consumes;
 import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
 import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
 import marquez.api.Job;
 import marquez.db.dao.JobDAO;
 
 @Path("/jobs")
 @Produces(MediaType.APPLICATION_JSON)
-@Consumes(MediaType.APPLICATION_JSON)
 public class JobResource {
-  private final JobDAO dao;
+  private final JobDAO jobDAO;
+
+  public JobResource(final JobDAO jobDAO) {
+    this.jobDAO = jobDAO;
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Timed
+  public Response create(final Job job) {
+    jobDAO.insert(job);
+    return Response.created(buildLocation(Job.class, job.getName())).build();
+  }
 
-  public JobResource(JobDAO dao) {
-    this.dao = dao;
+  @GET
+  @Timed
+  public Response getAll() {
+    return Response.ok(jobDAO.findAll()).build();
   }
 
   @GET
-  public List<Job> listJobs() {
-    return dao.findAll();
+  @Path("/{jobName}")
+  @Timed
+  public Response get(@PathParam("jobName") final String jobName) {
+    return Response.ok(jobDAO.findByName(jobName)).build();
```

Comment:
Could there be an error case to handle here? For instance, if there's no job found should it be an HTTP 404?

Reply:
For this and other endpoints?

Reply:
Yeah, we'll want to come up with a strategy to add general exception handling across all endpoints.